### DISCUSSION TITLE
Odd inconsistencies with WebC slots and HTML content

### DISCUSSION THREAD

**Discussion by cfjedimaster** [11ty/eleventy#2969] [Uncategorized]
I'm seeing some weirdness with a WebC component I built that uses a slot. When I called my component with basic stuff inside, so for example:

```
<bs-card title="Title is Ray" header="Header is here" subtitle="I'm a subtitle.">
	This is a card with all 3 attributes.
</bs-card>
```

It worked just fine. If I added a bit of HTML, it worked fine too:

```
<bs-card title="Title is Ray" header="Header is here" subtitle="I'm a subtitle.">
	This is a card with <strong>all 3</strong> attributes.
</bs-card>
```

But, as soon as I added some block level HTML, it "broke", such that the content was rendered, outside of the tag. For example:

```
<bs-card header="Awesome Header">
	<p>
	This card just has a header. I <strong>like</strong> this.
	</p>
</bs-card>
```

Here's my component:

```
<div class="card">
	<div class="card-header" webc:if="header" @text="header"></div>
	<div class="card-body">
		<h5 class="card-title" webc:if="title" @text="title"></h5>
		<h6 class="card-subtitle mb-2 text-body-secondary" webc:if="subtitle" @text="subtitle"></h6>
		<div class="card-text">
			<slot></slot>
		</div>
	</div>
</div>	
```

And rendered like so:

![image](https://github.com/11ty/eleventy/assets/393660/57c93b10-68b3-415d-9b4a-76f746e27e07)

Now, to make things weird, I decided to build a simpler reproducable case for foo. Here is foo.webc:

```

<h3>Foo Test</h3>
<div>
	<slot></slot>
</div>
<h3>End Foo Test</h3>

<style webc:scoped>
:host div {
	border-style:solid;
    border-width:thin;
	padding: 10px;
}
</style>
```

Same basic idea, right?  But if I do:

```
	<foo>
		<p>
		Para 1a
		</p>
		<hr>
		<p>
		Para 2
		</p>
	</foo>
```

It works perfectly.

**Comment by cfjedimaster** [comment]
So... I finally figured out why it happened. Well, how, not sure why. ;) When my test had this:

```html
<some-component>
html crap
</some-component>

<p>

<some-component>
another test html crap
</some-component>
```

It caused issues. What specifically? The unclosed P in the middle. Changing it to `<p></p>` corrected it.

**Reply by bennypowers** [reply]
this is possibly related to [#dsd](https://github.com/11ty/webc/issues/185), as markdown has a habit of injecting `<p>`s

**Reply by cfjedimaster** [reply]
Well in this case, I had entered P myself. So... my fault? But really weird how it impacted the component.
